homeassistant/components/media_player/browse_media.py

```diff
@@ -28,11 +30,15 @@ def async_process_play_media_url(
     for_supervisor_network: bool = False,
 ) -> str:
     """Update a media URL with authentication if it points at Home Assistant."""
-    if media_content_id[0] != "/" and not is_hass_url(hass, media_content_id):
-        return media_content_id
-
     parsed = yarl.URL(media_content_id)
 
+    if parsed.is_absolute():
+        if not is_hass_url(hass, media_content_id):
+            return media_content_id
+    else:
+        if media_content_id[0] != "/":
+            raise ValueError("URL is relative, but does not start with a /")
```

Comment:
If this happens, this method is used in the wrong context. It should only ever be passed things that are relative/absolute URLs